core/src/iam/jwks.rs

```diff
@@ -230,50 +242,42 @@ async fn fetch_jwks_from_url(url: &str) -> Result<JwkSet, Error> {
 	}
 }
 
-#[derive(Serialize, Deserialize)]
-struct JwksCache {
-	jwks: JwkSet,
-	time: DateTime<Utc>,
-}
-
 // Attempts to fetch a JWKS object from the local cache
-async fn fetch_jwks_from_cache(url: &str) -> Result<JwksCache, Error> {
-	let path = cache_path_from_url(url);
-	let bytes = crate::obs::get(&path).await?;
+async fn fetch_jwks_from_cache(
+	cache: &Arc<RwLock<JwksCache>>,
+	url: &str,
+) -> Option<JwksCacheEntry> {
+	let path = cache_key_from_url(url);
+	let cache_clone = Arc::clone(cache);
```

Comment:
If you are immediately cloning, you can take an owned Arc as a parameter. The clone on Arc is low-cost borrow counting - it doesnt clone the underlying object. So you can pass the owned value.